```
const log = i => {
    console.log(i, 'start');
    if (i + 1 < 10) {
        log(i + 1)
    }
    console.log(i,'end')
}
log(0)

0 "start"
1 "start"
2 "start"
3 "start"
4 "start"
5 "start"
6 "start"
7 "start"
8 "start"
9 "start"
9 "end"
8 "end"
7 "end"
6 "end"
5 "end"
4 "end"
3 "end"
2 "end"
1 "end"
0 "end"
```
因為log函數調用自身的時候，由於調用栈的原因沒有進入到第二個log，直接進入